src/ui/public/agg_types/buckets/histogram.js

```diff
@@ -3,13 +3,62 @@ import moment from 'moment';
 import 'ui/validate_date_interval';
 import AggTypesBucketsBucketAggTypeProvider from 'ui/agg_types/buckets/_bucket_agg_type';
 import AggTypesBucketsCreateFilterHistogramProvider from 'ui/agg_types/buckets/create_filter/histogram';
+import FilterBarQueryFilterProvider from 'ui/filter_bar/query_filter';
 import intervalTemplate from 'ui/agg_types/controls/interval.html';
 import minDocCountTemplate from 'ui/agg_types/controls/min_doc_count.html';
 import extendedBoundsTemplate from 'ui/agg_types/controls/extended_bounds.html';
+import autoScaleTemplate from 'ui/agg_types/controls/auto_scale.html';
 export default function HistogramAggDefinition(Private) {
   let BucketAggType = Private(AggTypesBucketsBucketAggTypeProvider);
   let createFilter = Private(AggTypesBucketsCreateFilterHistogramProvider);
+  let queryFilter = Private(FilterBarQueryFilterProvider);
+  const NUM_BUCKETS = 50;
 
+  function numberOfDigits(num) {
+    return (Math.trunc(num) + '').length;
+  }
+
+  function roundToNearest(num, roundDigit) {
+    if (roundDigit <= 0) {
+      return Math.round(num);
+    } else {
+      var transform = 10 * roundDigit;
+      return Math.round(num / transform) * transform;
+    }
+  }
+
+  function getInterval(agg) {
+    var interval = parseInt(agg.params.interval, 10);
+    if (agg.params.auto_scale) {
+      var found = false;
+      var min = 0;
+      var max = 0;
+      var key = agg.params.field.displayName;
+      _.flatten([queryFilter.getAppFilters()]).forEach(function (it) {
```

Comment:
`getAppFilters` won't include any pinned filters, you'll need to check `queryFilter.getGlobalFilters()` as well. Otherwise you could end up in a situation where a pinned filter has a small range, but the interval is based on a larger non-pinned filter and you'll get one big bucket like this:

![screen shot 2016-09-13 at 4 56 31 pm](https://cloud.githubusercontent.com/assets/6239176/18491286/a2c7d742-79d3-11e6-8011-c21b09b6fff4.png)